myhpom/tests/test_document_url_view.py

```diff
@@ -60,43 +56,36 @@ def test_get_document_url_succeeds(self):
     def test_document_url_404(self):
         """
         * accessing a DocumentUrl with past expiration returns 404
-        * accessing a DocumentUrl with correct key and incorrect filename returns 404
-        * accessing a DocumentURL not from authorized_client_ip returns 404
         * accessing a non-existent DocumentURL returns 404
         """
-        # wrong filename
-        doc_url = DocumentUrl.objects.create(advancedirective=self.__class__.advancedirective)
-        response = self.client.get(
-            reverse('myhpom:document_url', kwargs={'key': doc_url.key, 'filename': 'WRONG.pdf'})
-        )
-        self.assertEqual(404, response.status_code)
-
+        doc_url = DocumentUrl.objects.create(advancedirective=self.advancedirective)
         # wrong key
         wrong_key = 'W' * len(doc_url.key)  # right length, totally wrong value
         response = self.client.get(
-            reverse('myhpom:document_url', kwargs={'key': wrong_key, 'filename': doc_url.filename})
+            reverse('myhpom:document_url', kwargs={'key': wrong_key})
         )
         self.assertEqual(404, response.status_code)
 
-        # wrong ips
-        client_ip = response.wsgi_request.META['REMOTE_ADDR']
-        authorized_ips = [
-            "255.255.255.255",
-            "255.255.255.250, 255.255.255.255",
-            "255.255.255.255/24"
-        ]
-        for authorized_ip in authorized_ips:
-            self.assertNotEqual(client_ip, authorized_ip)
-            doc_url.ip = authorized_ip
-            doc_url.save()
-            response = self.client.get(doc_url.url)
-            self.assertLess(now(), doc_url.expiration)
-            self.assertEqual(404, response.status_code)
-
         # past expiration
-        doc_url.ip = None
         doc_url.expiration = now() - timedelta(seconds=1)
         doc_url.save()
         response = self.client.get(doc_url.url)
-        self.assertIsNone(doc_url.ip)
         self.assertEqual(404, response.status_code)
+
+    def test_document_url_not_authorized_ip(self):
+        """
+        * accessing a DocumentURL not from authorized_client_ip returns 404
+        """
+        doc_url = DocumentUrl.objects.create(advancedirective=self.advancedirective)
+        
+        # save the settings before changing them
+        save_settings = settings.DOCUMENT_URL_IP_RANGES
+
+        settings.DOCUMENT_URL_IP_RANGES = ["255.255.255.255"]  # pretty much no one, anywhere
+        response = self.client.get(doc_url.url)
+        self.assertLess(now(), doc_url.expiration)
+        self.assertEqual(404, response.status_code)
+
+        # restore settings
+        settings.DOCUMENT_URL_IP_RANGES = save_settings
```

Comment:
Use the `@override_settings` annotation to set `DOCUMENT_URL_IP_RANGES` rather than manually storing/saving/undoing.

Reply:
Perfect!